src/fsharp/FSharp.Core/Linq.fs

```diff
@@ -665,15 +636,45 @@ module LeafExpressionConverter =
     and failConvert inp =
             raise (new NotSupportedException(Printf.sprintf "Could not convert the following F# Quotation to a LINQ Expression Tree\n--------\n%A\n-------------\n" inp))
 
-    and transBinOp inp env addConvertLeft args addConvertRight (exprErasedConstructor : _ * _ -> _) =
-        match args with 
-        | [x1; x2] -> 
-            let e1 = ConvExprToLinqInContext env x1
-            let e2 = ConvExprToLinqInContext env x2
-            let e1 = if addConvertLeft  then Expression.Convert(e1, typedefof<Nullable<int>>.MakeGenericType [| e1.Type |]) |> asExpr else e1
-            let e2 = if addConvertRight then Expression.Convert(e2, typedefof<Nullable<int>>.MakeGenericType [| e2.Type |]) |> asExpr else e2
-            exprErasedConstructor(e1, e2) |> asExpr
-        | _ -> failConvert inp
+    and transUnaryOp inp env x (exprErasedConstructor: _ * _ -> _) fallback =
+        let e = ConvExprToLinqInContext env x
+        try exprErasedConstructor(e, null) with _ ->
```

Comment:
When do we expect this to raise exceptions?  I gather it's "LINQ Expressions' arithmetic operators do not handle byte, sbyte and char"

- In general we should not be adding catch-all exception handlers
- We should not be using exceptions for control flow and instead test before hand with an if/then/else

Also, I'm uneasy about this decision "LINQ Expressions' arithmetic operators do not handle byte, sbyte and char. In this case, use the F# operator as the user-defined method."   Do they really not accept byte?  That seems really wrong and we should possibly add a .NET bug about that.  

Either way this should be documented in the RFC as it's visible design choice.

Reply:
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Linq.Expressions/src/System/Dynamic/Utils/TypeUtils.cs#L110 exlucdes `Byte`, `SByte` and `Char`, and https://github.com/dotnet/runtime/blob/main/src/libraries/System.Linq.Expressions/README.md states that no feature additions will be accepted anymore.